test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs

```diff
@@ -65,5 +66,33 @@ public void WhenHelpOptionIsPassedToDotnetItPrintsUsage(string helpArg)
             cmd.Should().Pass();
             cmd.StdOut.Should().ContainVisuallySameFragment(HelpText);
         }
+
+        [Theory]
+        [InlineData("pack", "Usage: dotnet pack [arguments] [options] [args]")]
+        [InlineData("new", "Usage: dotnet new [arguments] [options]")]
+        [InlineData("restore", "Usage: dotnet restore [arguments] [options] [args]")]
+        [InlineData("build", "Usage: dotnet build [arguments] [options] [args]")]
+        [InlineData("publish", "Usage: dotnet publish [arguments] [options] [args]")]
+        [InlineData("run", "Usage: dotnet run [options] [[--] <additional arguments>...]]")]
+        [InlineData("test", "Usage: dotnet test [arguments] [options] [args]")]
+        [InlineData("migrate", "Usage: dotnet migrate [arguments] [options]")]
+        [InlineData("clean", "Usage: dotnet clean [arguments] [options] [args]")]
+        [InlineData("sln", "Usage: dotnet sln [arguments] [options] [command]")]
+        [InlineData("sln add", "Usage: dotnet sln <SLN_FILE> add [options] [args]")]
+        [InlineData("sln list", "Usage: dotnet sln <SLN_FILE> list [options]")]
+        [InlineData("sln remove", "Usage: dotnet sln <SLN_FILE> remove [options] [args]")]
+        [InlineData("nuget", "Usage: dotnet nuget [options] [command]")]
+        // [InlineData("msbuild", "")]
```

Comment:
#4813 mentions that there's also an issue with `dotnet msbuild -h`. The output there is quite inconsistent with the others. I included the commented out code as a placeholder pending further discussion. Is this something we want to try to address now?

```
$ dotnet msbuild -h
Microsoft (R) Build Engine version 15.1.545.13942
Copyright (C) Microsoft Corporation. All rights reserved.

Syntax:              MSBuild.exe [options] [project file | directory]

Description:         Builds the specified targets in the project file. If
                     a project file is not specified, MSBuild searches the
                     current working directory for a file that has a file
                     extension that ends in "proj" and uses that file.  If
                     a directory is specified, MSBuild searches that
                     directory for a project file.

...
```